templates/mount_vm.sh.j2

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# {{ ansible_managed }}
+
+function mount_iscsi() {
+	local blk_dev="/dev/disk/by-path/ip-{{ xen_vman_iscsi_server }}-iscsi-{{ xen_vman_iscsi_base_wwn }}:$vm-lun-0"
+	# login into iscsi for requested vm, needed for standalone starts(via script), do not so if started via systemd
+	[ ! -b "$blk_dev" -a -z "$INVOCATION_ID" ] && {
+		touch /tmp/.$vm.iscsi_started_by_mount_script
+		systemctl start vm_iscsi@$vm.service
+  	}
+	timeout=$(({{ xen_vman_iscsi_login_timeout }}*2))
+	while [ ! -b "$blk_dev" ]; do  # wait for block device
+	  sleep 0.5
+	  timeout=$((timeout-1))
+	done
+	[ "$timeout" -le 0 ] && { echo "ERROR:   Iscsi login has reached timeout"; exit 1; }
+	mount -v -o "defaults,rw" "$blk_dev" "$mount_path"
+}
+
+function mount_nfs() {
+	mount -v "{{ xen_vman_nfsroot_base }}$vm-root" "$mount_path"
+}
+
+function detect_storage_type() {
+	sed -n 's/# storage type:[ ]\+//p' "$config_path"
+}
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
```

Comment:
No getopt?

Reply:
Yup, getopt has to limited features and or adds to much complexity.